CMakeLists.txt

```diff
@@ -38,14 +38,23 @@ vlk_get_header_version()
 
 project(VULKAN_HEADERS LANGUAGES C VERSION ${VK_VERSION_STRING})
 
-add_library(Vulkan-Headers INTERFACE)
+add_library(Vulkan-Headers)
 add_library(Vulkan::Headers ALIAS Vulkan-Headers)
-target_include_directories(Vulkan-Headers INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+target_sources(Vulkan-Headers
+    PUBLIC
+        FILE_SET headers TYPE HEADERS
+            BASE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/include"
+)
 
-if (CMAKE_VERSION VERSION_LESS "3.21")
-    # https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html
-    string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} PROJECT_IS_TOP_LEVEL)
-endif()
+add_library(Vulkan-Module)
+add_library(Vulkan::VulkanHppModule ALIAS Vulkan-Module)
+target_sources(Vulkan-Module
+    PUBLIC
+        FILE_SET module CXX_MODULES
+            BASE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/include"
+            FILES "${CMAKE_CURRENT_SOURCE_DIR}/include/vulkan.cppm"
```

Comment:
Use of CMake features this new should be guarded by if-checks, like the one seen on line 45 for CMake 3.21.

Reply:
Thanks for the review. Would 3.23 be more acceptable? This is the earliest version that introduces `FILE_SET`, which would also allow dropping the test in line 45.

Reply:
Cheers. I've guarded this behind a similar `if` statement, and restored line 45 itself.